src/hotspot/share/runtime/thread.inline.hpp

```diff
@@ -91,4 +91,27 @@ inline WXMode Thread::enable_wx(WXMode new_state) {
 }
 #endif // __APPLE__ && AARCH64
 
+
+#ifdef ASSERT
+// Mark disables Thread::current by setting it to NULL for the extend. Can be
+// used to simulate running code in non-attached threads, or to guard code
+// against accidental usage of features that depend on Thread::current() (e.g.
```

Comment:
So the "guard" aspect of this is that we expect to crash if the code tries to use the result of `Thread::current()`?